### PR TITLE
Rework aircraft rally point handling.

### LINKED ISSUE #16912: Airfield rally point not working
Produced planes don't move to the airfield's rally point and idle after take-off. Tested on 7cfc65010f9eb3a5f8f3b4145e918fa8f2419f8f.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #16912 and hopefully makes the code a bit more sensible and understandable.

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Resupply.cs [hunk 1]
```diff
@@ -169,12 +169,22 @@ public override IEnumerable<TargetLineNode> TargetLineNodes(Actor self)
...
 		void OnResupplyEnding(Actor self, bool isHostInvalid = false)
 		{
+			var rp = host.Actor.TraitOrDefault<RallyPoint>();
 			if (aircraft != null)
 			{
-				aircraft.AllowYieldingReservation();
-				if (wasRepaired || isHostInvalid ||
-					(!stayOnResupplier && aircraft.Info.TakeOffOnResupply))
-					QueueChild(new TakeOff(self));
+				// Aircraft remain on the resupplier until something els
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Activities/Resupply.cs]: This comment looks unfinished...

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs [hunk 2]
```diff
@@ -521,13 +521,21 @@ public void UnReserve(bool takeOff = false)
...
 			reservation.Dispose();
 			reservation = null;
 			ReservedActor = null;
 			MayYieldReservation = false;
 
 			if (takeOff && self.World.Map.DistanceAboveTerrain(CenterPosition).Length <= LandAltitude.Length)
-				self.QueueActivity(new TakeOff(self));
+			{
+				if (rp != null)
+					self.QueueActivity(new TakeOff(self, Target.FromCell(self.World, rp.Location)));
```
Comment (tovl) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Why call `TakeOff` here but `move.MoveTo` in `Resupply`?
If you call  `move.MoveTo` here as well (optionally wrapped in `AttackMoveActivity` if that is indeed the desired behaviour) then you can get rid of the target handling in `TakeOff` completely.

Comment (pchote) [reply OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: That was originally my plan, but then I realized that `TakeOff` with a target is not equivalent to `move.MoveTo` when there are activities queued afterwards. Trying to untangle and rewrite the different cases vastly increases the scope and regression potential of the PR.

The `move.MoveTo` in `Resupply` is copied directly from the `Mobile` branch, with the goal of having similar behaviour for all units.

**Comment by pchote** [comment]
Fixed and rebased.

**Review by ghost — APPROVED** [review]
LGTM.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/25e14ee4319185cd644746ab48690b0e711bb1f3...a77d5335f2d524ba97d0f62c77c52aa246525285

**Pull request merged**